msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/DiscussionForumTool.java

```diff
@@ -2224,6 +2236,220 @@ public String processActionToggleDisplayExtendedDescription()
+      calendarBeginId = ((DiscussionTopic) forumItem).getCalendarBeginId();
+      calendarEndId = ((DiscussionTopic) forumItem).getCalendarEndId();
+      sendOpenCloseToCalendar = ((DiscussionTopic) forumItem).getSendOpenCloseToCalendar();
+      openDate = ((DiscussionTopic) forumItem).getOpenDate();
+      closeDate = ((DiscussionTopic) forumItem).getCloseDate();
+      membershipItems = ((DiscussionTopic) forumItem).getMembershipItemSet();
+      availabilityRestrictedNow = ((DiscussionTopic) forumItem).getAvailabilityRestricted();
+      openingTitle = getResourceBundleString("sendOpenCloseToCalendar.title.topic.assembly",new Object[]{((DiscussionTopic) forumItem).getTitle(),getResourceBundleString("sendOpenCloseToCalendar.topic.open")});
+      closingTitle = getResourceBundleString("sendOpenCloseToCalendar.title.topic.assembly",new Object[]{((DiscussionTopic) forumItem).getTitle(),getResourceBundleString("sendOpenCloseToCalendar.topic.close")});
+    } else if (forumItem instanceof DiscussionForum) {
+      calendarBeginId = ((DiscussionForum) forumItem).getCalendarBeginId();
+      calendarEndId = ((DiscussionForum) forumItem).getCalendarEndId();
+      sendOpenCloseToCalendar = ((DiscussionForum) forumItem).getSendOpenCloseToCalendar();
+      openDate = ((DiscussionForum) forumItem).getOpenDate();
+      closeDate = ((DiscussionForum) forumItem).getCloseDate();
+      membershipItems = ((DiscussionForum) forumItem).getMembershipItemSet();
+      availabilityRestrictedNow = ((DiscussionForum) forumItem).getAvailabilityRestricted();
+      openingTitle = getResourceBundleString("sendOpenCloseToCalendar.title.forum.assembly",new Object[]{((DiscussionForum) forumItem).getTitle(),getResourceBundleString("sendOpenCloseToCalendar.topic.open")});
+      closingTitle = getResourceBundleString("sendOpenCloseToCalendar.title.forum.assembly",new Object[]{((DiscussionForum) forumItem).getTitle(),getResourceBundleString("sendOpenCloseToCalendar.topic.close")});
+    } else {  //don't do anything with forumItem or the Calendar if it's not a Forum or a Forum Topic.
+      return;
+    }
+    Collection<Group> allowedGroups = getAllowedGroups(membershipItems);
+    if (allowedGroups.size() == 0 && !studentsAllowed(membershipItems)){
+      sendOpenCloseToCalendar = false; //if no groups or students are allowed to see this, we will treat it as though Send is not checked at all.
+      if (forumItem instanceof DiscussionTopic){  //clear the forum item's value as well, if this is the case.
+        ((DiscussionTopic) forumItem).setSendOpenCloseToCalendar(false);
+      } else if (forumItem instanceof DiscussionForum) {
+        ((DiscussionForum) forumItem).setSendOpenCloseToCalendar(false);
+      }
+    }
+    try {   //now actually start processing the data for Calendar.
+      Calendar targetCalendar = this.calendarService.getCalendar("/calendar/calendar/" + getContextSiteId().replace("/site/", "") + "/main");
```

Comment:
```suggestion
      Calendar targetCalendar = this.calendarService.getCalendar("/calendar/calendar/" + getContextSiteId().replace("/site/", "") + "/main");
      Calendar targetCalendar = calendarService.calendarReference(getContextSiteId(), siteService.MAIN_CONTAINER);
```

Reply:
Did you mean to leave line 2283 in there and follow it with 2284, or is 2284 meant to be a replacement? I'm pretty sure it's a replacement but I just want to make sure before I commit it.

Reply:
Just went into BaseCalendarService.calendarReference to check this out, and I'm pretty sure this line should end up like this: `Calendar targetCalendar = this.calendarService.getCalendar(calendarService.calendarReference(getContextSiteId(), siteService.MAIN_CONTAINER));`